### PR TITLE
Fixes #15 | Restarting apps metrics

### LINKED ISSUE #15: Application restarts metrics
`AppCop` should provide metric of each application status change (TASK_RUNNING, TASK_STAGING, TASK_KILLED) in the form of graphite metrics, based on events.

### PR DESCRIPTION
**Pull request by tomez**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,5 @@ marathon-forwarder
 
 # coverage
 /coverage
+
+tags
```
Comment (janisz) [inline .gitignore:39]: Who creates this file?

File: README.md [hunk 2]
```diff
@@ -95,6 +120,7 @@ marathon-password           |                   | Marathon password for basic au
 marathon-protocol           | `http`            | Marathon protocol (http or https)
 marathon-ssl-verify         | `true`            | Verify certificates when connecting via SSL
 marathon-timeout            | `30s`             | Time limit for requests made by the Marathon HTTP client. A timeout of zero means no timeout
+appid-prefix                |                   | Prefix common to all fully qualified application ID's. Remove this preffix from applications id's (reffer to README to get an idea when this id is removed)
```
Comment (janisz) [inline README.md]: It's in readme. Add link to specific section

File: config/config.go [hunk 3]
```diff
@@ -122,10 +124,16 @@ func (config *Config) parseFlags() {
 		"Metrics destination stdout or graphite (empty string disables metrics)")
 	flag.StringVar(&config.Metrics.Prefix, "metrics-prefix", "default",
 		"Metrics prefix (default is resolved to <hostname>.<app_name>")
+	flag.StringVar(&config.Metrics.SystemSubPrefix, "metrics-system-sub-prefix", "appcop-internal",
+		"System specific metrics. Append t o metric-prefix")
```
Comment (janisz) [inline config/config.go]: `t o` → `to`

File: README.md [hunk 4]
```diff
@@ -104,6 +130,8 @@ evaluate-interval           | `30s`             | How often collected scores are
 metrics-interval            | `30s`             | Metrics reporting interval
 metrics-location            |                   | Graphite URL (used when metrics-target is set to graphite)
 metrics-prefix              | `default`         | Metrics prefix (default is resolved to <hostname>.<app_name>
+metrics-system-sub-prefix   | `appcop-internal` | System specific metrics. Append t o metric-prefix
```
Comment (janisz) [inline README.md]: `t o` → `to`

File: marathon/app.go [hunk 5]
```diff
@@ -96,6 +98,31 @@ type Task struct {
...
+	noRootAppID := dropRootSlashID(appID)
+	metricPath := strings.Replace(noRootAppID, metrics.PathSeparator, metrics.MetricSeparator, -1)
+	taskStatus := strings.ToLower(t.TaskStatus)
+
+	return strings.Join([]string{metricPath, taskStatus}, metrics.MetricSeparator)
+
+}
+
+// dropRootSlashDI drops a beginning '/' from appID
+func dropRootSlashID(data string) string {
```
Comment (janisz) [inline marathon/app.go]: I'd prefer [`strings.TrimPrefix`](https://golang.org/pkg/strings/#TrimPrefix) instead of this.

File: marathon/app.go [hunk 6]
```diff
@@ -96,6 +98,31 @@ type Task struct {
 	HealthCheckResults []HealthCheckResult `json:"healthCheckResults"`
 }
 
+func (t Task) GetMetric(prefix string) string {
```
Comment (janisz) [inline marathon/app.go]: It's not obvious what this function does. Please add a comment.

File: marathon/app_test.go [hunk 7]
```diff
@@ -93,6 +93,85 @@ func TestParseTasksRecievesMalformedJSONBlob(t *testing.T) {
...
+			AppID:      "app-name",
+		},
+		prefix:             "",
+		expectedTaskMetric: "app-name.task_running",
+	},
+	{
+		task: &Task{
+			TaskStatus: "task_running",
+			AppID:      "com.example.domain.context/app-name",
+		},
```
Comment (janisz) [inline marathon/app_test.go:169]: Add test with empty TaskStatus and AppId

File: metrics/config.go [hunk 8]
```diff
@@ -4,8 +4,11 @@ import "time"
...
-	Prefix   string
-	Interval time.Duration
-	Addr     string
+	Target          string
+	Prefix          string
+	Interval        time.Duration
+	Addr            string
+	Instance        string
+	SystemSubPrefix string
+	AppSubPrefix    string
```
Comment (janisz) [inline metrics/config.go]: It's good time to add docs at least to the new fileds.

File: marathon/marathon.go [hunk 9]
```diff
@@ -81,10 +83,11 @@ func New(config Config) (*Marathon, error) {
...
 	return &Marathon{
-		Location: config.Location,
-		Protocol: config.Protocol,
-		Auth:     auth,
-		client:   pClient,
+		Location:    config.Location,
+		Protocol:    config.Protocol,
+		appIDPrefix: config.AppIDPrefix,
+		Auth:        auth,
+		client:      pClient,
```
Comment (janisz) [inline marathon/marathon.go:90]: It's good time to add docs at least to the new fileds.

**Review by janisz — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (tomez) [reply .gitignore:39]: https://github.com/jstemmer/gotags - ctags for project, used by IDE.

**Review by janisz — APPROVED** [review]
